compiler/src/dotty/tools/dotc/tastyreflect/TastyImpl.scala

```diff
@@ -538,7 +538,7 @@ object TastyImpl extends scala.tasty.Tasty {
 
     object Synthetic extends SyntheticExtractor {
       def unapply(x: TypeTree)(implicit ctx: Context): Boolean = x match {
-        case Trees.TypeTree() => true
+        case x @ Trees.TypeTree() => !x.tpe.isInstanceOf[Types.TypeBounds]
         case _ => false
       }
     }
```

Comment:
Given the name `synthetic`, I cannot justify the change. Maybe a doc for `Synthetic`, I naively assume synthetic bounds are also synthetic.

Reply:
I will add some documentation.

Reply:
Added